python/cudf/cudf/tests/test_json.py

```diff
@@ -618,3 +618,16 @@ def test_json_nested_lines(data):
     # In the second test-case:
     # Pandas omits "f1" in first row, so we have to enforce a common schema
     assert df.to_arrow().equals(pa.Table.from_pandas(pdf))
+
+
+def test_json_nested_data():
```

Comment:
hm, would it be easier to create a dataframe, write to JSON with Pandas and compare the read dataframe with the original?

Reply:
We would still have to use `pyarrow.equals` method to compare nested data and I usually found pandas assert_eq seems to be not so good at comparing nested data. So we will have to end up doing the same workarounds because of the ordering shenanigans.